core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -391,6 +391,15 @@ public IndexResult(long version, long seqNo, boolean created) {
             this.created = created;
         }
 
+        /**
+         * use in case of index operation failed before getting to internal engine
+         * (e.g while preparing operation or updating mappings)
+         * */
+        public IndexResult(Exception failure, long version) {
+            super(Operation.TYPE.INDEX, failure, version, SequenceNumbersService.UNASSIGNED_SEQ_NO);
```

Comment:
I think this would be better as:

```java
public IndexResult(Exception failure, long version) {
    this(failure, version, SequenceNumbersService.UNASSIGNED_SEQ_NO);
}
```

Since it collapses the constructor logic into fewer places